---
title: "Feature Selection and Clustering of PBMC UMI Data"
output: rmarkdown::html_vignette
vignette: >
  %\VignetteIndexEntry{Example analysis}
  %\VignetteEngine{knitr::rmarkdown}
  \usepackage[utf8]{inputenc}
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE,
                      message = FALSE,
                      warning = FALSE,
                      dpi = 150,
                      comment = NA)
library(HIPPO)
library(ggplot2)
library(Matrix)
```

## Read the data

The data set is available in the following [link](http://imlspenticton.uzh.ch/robinson_lab/DuoClustering2018/DuoClustering2018.tar.gz), where the detailed explanation is available [here](https://github.com/markrobinsonuzh/scRNAseq_clustering_comparison). We use Zhengmix4eq data set. From the SingleCellExperiment object, we extracted the count matrix only (gene x cell) as .rds file, and we also created the reference matrix where the gene names are matched between HGNC symbol and ENSG IDs. 

```{r}
X = readRDS("../zhengmix4eq_counts.rds")
ref = read.table("../ensg_hgnc_reference.txt", header=TRUE)
```

## Feature Selection and Hierarchical Clustering

Below, hippo_object will return each round of feature selection and clustering result. K = 4 and z-threshold = 5 is the stopping criterion for hierarchical clustering. The algorithm will automatically stop after 4 clustering or if there are no more features left with z-scores higher than 5.

```{r}
set.seed(20191031)
hippo_object = hippo(X, K = 4, z_threshold = 5)
```

## Visualize the selected features at each round

This shows the UMAP and t-SNE plot of each round of clustering. 

```{r}
hippo_plots = visualize_hippo(hippo_object)
```

```{r, fig.width=6, fig.height=4}
hippo_plots$zero_plot
hippo_plots$umap_plot
hippo_plots$tsne_plot
```

## Differential Expression Example

```{r, fig.width=6, fig.height=4}
out = diffexp(hippo_object, top.n = 5, switch_to_hgnc = TRUE, ref = ref)
```

```{r}
sessionInfo()
```
