<!DOCTYPE html>

<html>

<head>

<meta charset="utf-8" />
<meta name="generator" content="pandoc" />
<meta http-equiv="X-UA-Compatible" content="IE=EDGE" />




<title>Example2: PBMC68K</title>

<script src="site_libs/jquery-1.11.3/jquery.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/cosmo.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<link href="site_libs/highlightjs-9.12.0/textmate.css" rel="stylesheet" />
<script src="site_libs/highlightjs-9.12.0/highlight.js"></script>

<style type="text/css">code{white-space: pre;}</style>
<style type="text/css">
  pre:not([class]) {
    background-color: white;
  }
</style>
<script type="text/javascript">
if (window.hljs) {
  hljs.configure({languages: []});
  hljs.initHighlightingOnLoad();
  if (document.readyState && document.readyState === "complete") {
    window.setTimeout(function() { hljs.initHighlighting(); }, 0);
  }
}
</script>



<style type="text/css">
h1 {
  font-size: 34px;
}
h1.title {
  font-size: 38px;
}
h2 {
  font-size: 30px;
}
h3 {
  font-size: 24px;
}
h4 {
  font-size: 18px;
}
h5 {
  font-size: 16px;
}
h6 {
  font-size: 12px;
}
.table th:not([align]) {
  text-align: left;
}
</style>




<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
code {
  color: inherit;
  background-color: rgba(0, 0, 0, 0.04);
}
img {
  max-width:100%;
}
.tabbed-pane {
  padding-top: 12px;
}
.html-widget {
  margin-bottom: 20px;
}
button.code-folding-btn:focus {
  outline: none;
}
summary {
  display: list-item;
}
</style>


<style type="text/css">
/* padding for bootstrap navbar */
body {
  padding-top: 51px;
  padding-bottom: 40px;
}
/* offset scroll position for anchor links (for fixed navbar)  */
.section h1 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h2 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h3 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h4 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h5 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h6 {
  padding-top: 56px;
  margin-top: -56px;
}
.dropdown-submenu {
  position: relative;
}
.dropdown-submenu>.dropdown-menu {
  top: 0;
  left: 100%;
  margin-top: -6px;
  margin-left: -1px;
  border-radius: 0 6px 6px 6px;
}
.dropdown-submenu:hover>.dropdown-menu {
  display: block;
}
.dropdown-submenu>a:after {
  display: block;
  content: " ";
  float: right;
  width: 0;
  height: 0;
  border-color: transparent;
  border-style: solid;
  border-width: 5px 0 5px 5px;
  border-left-color: #cccccc;
  margin-top: 5px;
  margin-right: -10px;
}
.dropdown-submenu:hover>a:after {
  border-left-color: #ffffff;
}
.dropdown-submenu.pull-left {
  float: none;
}
.dropdown-submenu.pull-left>.dropdown-menu {
  left: -100%;
  margin-left: 10px;
  border-radius: 6px 0 6px 6px;
}
</style>

<script>
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark it active
  menuAnchor.parent().addClass('active');

  // if it's got a parent navbar menu mark it active as well
  menuAnchor.closest('li.dropdown').addClass('active');
});
</script>

<!-- tabsets -->

<style type="text/css">
.tabset-dropdown > .nav-tabs {
  display: inline-table;
  max-height: 500px;
  min-height: 44px;
  overflow-y: auto;
  background: white;
  border: 1px solid #ddd;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs > li.active:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li.active:before {
  content: "&#xe258;";
  border: none;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs > li.active {
  display: block;
}

.tabset-dropdown > .nav-tabs > li > a,
.tabset-dropdown > .nav-tabs > li > a:focus,
.tabset-dropdown > .nav-tabs > li > a:hover {
  border: none;
  display: inline-block;
  border-radius: 4px;
  background-color: transparent;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li {
  display: block;
  float: none;
}

.tabset-dropdown > .nav-tabs > li {
  display: none;
}
</style>

<!-- code folding -->




</head>

<body>


<div class="container-fluid main-container">




<div class="navbar navbar-default  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">HIPPO</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="index.html">Home</a>
</li>
<li>
  <a href="about.html">About</a>
</li>
<li>
  <a href="example.html">Example1</a>
</li>
<li>
  <a href="example2.html">Example2</a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<!-- Add a small amount of space between sections. -->
<style type="text/css">
div.section {
  padding-top: 12px;
}
</style>

<div class="fluid-row" id="header">



<h1 class="title toc-ignore">Example2: PBMC68K</h1>

</div>


<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-report" data-toggle="collapse" data-target="#workflowr-report">
<span class="glyphicon glyphicon-list" aria-hidden="true"></span> workflowr <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span>
</button>
</p>
<div id="workflowr-report" class="collapse">
<ul class="nav nav-tabs">
<li class="active">
<a data-toggle="tab" href="#summary">Summary</a>
</li>
<li>
<a data-toggle="tab" href="#checks"> Checks <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> </a>
</li>
<li>
<a data-toggle="tab" href="#versions">Past versions</a>
</li>
</ul>
<div class="tab-content">
<div id="summary" class="tab-pane fade in active">
<p>
<strong>Last updated:</strong> 2020-01-17
</p>
<p>
<strong>Checks:</strong> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> 7 <span class="glyphicon glyphicon-exclamation-sign text-danger" aria-hidden="true"></span> 0
</p>
<p>
<strong>Knit directory:</strong> <code>HIPPO/</code> <span class="glyphicon glyphicon-question-sign" aria-hidden="true" title="This is the local directory in which the code in this file was executed."> </span>
</p>
<p>
This reproducible <a href="http://rmarkdown.rstudio.com">R Markdown</a> analysis was created with <a
  href="https://github.com/jdblischak/workflowr">workflowr</a> (version 1.5.0). The <em>Checks</em> tab describes the reproducibility checks that were applied when the results were created. The <em>Past versions</em> tab lists the development history.
</p>
<hr>
</div>
<div id="checks" class="tab-pane fade">
<div id="workflowr-checks" class="panel-group">
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRMarkdownfilestronguptodate"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>R Markdown file:</strong> up-to-date </a>
</p>
</div>
<div id="strongRMarkdownfilestronguptodate" class="panel-collapse collapse">
<div class="panel-body">
<p>Great! Since the R Markdown file has been committed to the Git repository, you know the exact version of the code that produced these results.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongEnvironmentstrongempty"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Environment:</strong> empty </a>
</p>
</div>
<div id="strongEnvironmentstrongempty" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! The global environment was empty. Objects defined in the global environment can affect the analysis in your R Markdown file in unknown ways. For reproduciblity it’s best to always run the code in an empty environment.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSeedstrongcodesetseed20200117code"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Seed:</strong> <code>set.seed(20200117)</code> </a>
</p>
</div>
<div id="strongSeedstrongcodesetseed20200117code" class="panel-collapse collapse">
<div class="panel-body">
<p>The command <code>set.seed(20200117)</code> was run prior to running the code in the R Markdown file. Setting a seed ensures that any results that rely on randomness, e.g. subsampling or permutations, are reproducible.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSessioninformationstrongrecorded"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Session information:</strong> recorded </a>
</p>
</div>
<div id="strongSessioninformationstrongrecorded" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Recording the operating system, R version, and package versions is critical for reproducibility.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongCachestrongnone"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Cache:</strong> none </a>
</p>
</div>
<div id="strongCachestrongnone" class="panel-collapse collapse">
<div class="panel-body">
<p>Nice! There were no cached chunks for this analysis, so you can be confident that you successfully produced the results during this run.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongFilepathsstrongrelative"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>File paths:</strong> relative </a>
</p>
</div>
<div id="strongFilepathsstrongrelative" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Using relative paths to the files within your workflowr project makes it easier to run your code on other machines.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRepositoryversionstrongahrefhttpsgithubcomtk382HIPPOtreeb78600d87357063b7b5b56f3a70876c2ab1fa2aftargetblankb78600da"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Repository version:</strong> <a href="https://github.com/tk382/HIPPO/tree/b78600d87357063b7b5b56f3a70876c2ab1fa2af" target="_blank">b78600d</a> </a>
</p>
</div>
<div id="strongRepositoryversionstrongahrefhttpsgithubcomtk382HIPPOtreeb78600d87357063b7b5b56f3a70876c2ab1fa2aftargetblankb78600da" class="panel-collapse collapse">
<div class="panel-body">
<p>
Great! You are using Git for version control. Tracking code development and connecting the code version to the results is critical for reproducibility. The version displayed above was the version of the Git repository at the time these results were generated. <br><br> Note that you need to be careful to ensure that all relevant files for the analysis have been committed to Git prior to generating the results (you can use <code>wflow_publish</code> or <code>wflow_git_commit</code>). workflowr only checks the R Markdown file, but you know if there are other scripts or data files that it depends on. Below is the status of the Git repository when the results were generated:
</p>
<pre><code>
Ignored files:
    Ignored:    .Rhistory
    Ignored:    .Rproj.user/
    Ignored:    R/.Rhistory

Untracked files:
    Untracked:  make_sce_zhengmix4eq.R
    Untracked:  sce_full_Zhengmix4eq.rds

</code></pre>
<p>
Note that any generated files, e.g. HTML, png, CSS, etc., are not included in this status report because it is ok for generated content to have uncommitted changes.
</p>
</div>
</div>
</div>
</div>
<hr>
</div>
<div id="versions" class="tab-pane fade">

<p>
These are the previous versions of the R Markdown and HTML files. If you’ve configured a remote Git repository (see <code>?wflow_git_remote</code>), click on the hyperlinks in the table below to view them.
</p>
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
File
</th>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
<th>
Message
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/tk382/HIPPO/blob/b78600d87357063b7b5b56f3a70876c2ab1fa2af/analysis/example2.Rmd" target="_blank">b78600d</a>
</td>
<td>
tk382
</td>
<td>
2020-01-17
</td>
<td>
wflow_publish(c(“analysis/example.Rmd”, “analysis/example2.Rmd”,
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/tk382/HIPPO/blob/b404eaefa44b5c24a761b7d5e09c10bf821a8eda/analysis/example2.Rmd" target="_blank">b404eae</a>
</td>
<td>
tk382
</td>
<td>
2020-01-17
</td>
<td>
add workflow analysis files
</td>
</tr>
</tbody>
</table>
</div>
<hr>
</div>
</div>
</div>
<div id="read-the-data" class="section level2">
<h2>Read the data</h2>
<p>This data set is available in 10X <a href="https://support.10xgenomics.com/single-cell-gene-expression/datasets/1.1.0/fresh_68k_pbmc_donor_a">website</a>. The format is Matrix, so we need Matrix package for this particular example.</p>
<pre class="r"><code>library(Matrix)
X = readMM(&quot;/Volumes/tae/Work/SC/data/10X_68K/filtered_matrix/hg19/matrix.mtx&quot;)
rowdata =  read.table(&quot;/Volumes/tae/Work/SC/data/10X_68K/filtered_matrix/hg19/genes.tsv&quot;) # this shows both ENSG IDs and HGNC symbols
coldata = data.table::setDF(data.table::fread(&quot;/Volumes/tae/Work/SC/data/10X_68K/filtered_matrix/hg19/barocdes_annotation.tsv&quot;)) # this shows barcode information
rownames(X) = rowdata$V1
colnames(X) = coldata$barcodes</code></pre>
<p>My local laptop cannot handle such a large matrix, so I will reduce the number of cells by random sampling.</p>
<pre class="r"><code>set.seed(20200117)
cells = sample(1:ncol(X), size = 3000)
X = X[, cells]
coldata = coldata[cells, ]</code></pre>
<p>Then create the SingleCellExperiment object.</p>
<pre class="r"><code>sce = SingleCellExperiment(assays = list(counts = as.matrix(X)),
                           rowData = rowdata,
                           colData = coldata)

rm(X, rowdata, coldata); gc()</code></pre>
<pre><code>            used  (Mb) gc trigger   (Mb) limit (Mb)  max used   (Mb)
Ncells   5399381 288.4   10211220  545.4         NA   6410034  342.4
Vcells 107968179 823.8  216520851 1652.0      16384 180070227 1373.9</code></pre>
</div>
<div id="diagnostic-plot" class="section level2">
<h2>Diagnostic Plot</h2>
<pre class="r"><code>hippo_diagnostic_plot(sce)</code></pre>
<p><img src="figure/example2.Rmd/unnamed-chunk-4-1.png" width="672" style="display: block; margin: auto;" /></p>
</div>
<div id="feature-selection-and-hierarchical-clustering" class="section level2">
<h2>Feature Selection and Hierarchical Clustering</h2>
<pre class="r"><code>set.seed(20191031)
sce = hippo(sce, K = 10, z_threshold = 2, outlier_proportion = 0.01)</code></pre>
<pre><code>[1] &quot;K = 2..&quot;
[1] &quot;K = 3..&quot;
[1] &quot;K = 4..&quot;
[1] &quot;not enough important features left; terminating the procedure&quot;</code></pre>
</div>
<div id="dimension-reduction-for-each-round-of-hippo" class="section level2">
<h2>Dimension Reduction for Each Round of HIPPO</h2>
<pre class="r"><code>sce = dimension_reduction(sce, method=&quot;umap&quot;)
hippo_umap_plot(sce)</code></pre>
<p><img src="figure/example2.Rmd/unnamed-chunk-6-1.png" width="480" style="display: block; margin: auto;" /><img src="figure/example2.Rmd/unnamed-chunk-6-2.png" width="480" style="display: block; margin: auto;" /></p>
<pre class="r"><code>sce = dimension_reduction(sce, method=&quot;tsne&quot;)
hippo_tsne_plot(sce)</code></pre>
<p><img src="figure/example2.Rmd/unnamed-chunk-7-1.png" width="576" style="display: block; margin: auto;" /><img src="figure/example2.Rmd/unnamed-chunk-7-2.png" width="576" style="display: block; margin: auto;" /></p>
</div>
<div id="visualize-the-selected-features-at-each-round" class="section level2">
<h2>Visualize the selected features at each round</h2>
<pre class="r"><code>zero_proportion_plot(sce)</code></pre>
<p><img src="figure/example2.Rmd/unnamed-chunk-8-1.png" width="576" style="display: block; margin: auto;" /><img src="figure/example2.Rmd/unnamed-chunk-8-2.png" width="576" style="display: block; margin: auto;" /></p>
</div>
<div id="differential-expression-example" class="section level2">
<h2>Differential Expression Example</h2>
<pre class="r"><code>ref = data.frame(hgnc = rowData(sce)$V2,
                 ensg = rowData(sce)$V1)
head(ref)</code></pre>
<pre><code>          hgnc            ensg
1   MIR1302-10 ENSG00000243485
2      FAM138A ENSG00000237613
3        OR4F5 ENSG00000186092
4 RP11-34P13.7 ENSG00000238009
5 RP11-34P13.8 ENSG00000239945
6   AL627309.1 ENSG00000237683</code></pre>
<pre class="r"><code>sce = diffexp(sce, top.n = 5, switch_to_hgnc = TRUE, ref = ref)</code></pre>
<p><img src="figure/example2.Rmd/unnamed-chunk-9-1.png" width="576" style="display: block; margin: auto;" /></p>
<pre class="r"><code>head(sce@int_metadata$hippo$diffexp$result_table[[1]]) #round 1: monocytes</code></pre>
<pre><code>              genes  meandiff        sd        z     hgnc
254 ENSG00000019582 17.038462 0.2134361 79.82935     CD74
908 ENSG00000087086 15.955349 0.2358395 67.65342      FTL
295 ENSG00000204287  9.472679 0.1564456 60.54933  HLA-DRA
507 ENSG00000167996 11.558238 0.2021407 57.17916     FTH1
297 ENSG00000196126  7.187445 0.1375766 52.24321 HLA-DRB1
307 ENSG00000223865  5.669290 0.1227722 46.17730 HLA-DPB1</code></pre>
<pre class="r"><code>head(sce@int_metadata$hippo$diffexp$result_table[[2]]) #round 2: B cells</code></pre>
<pre><code>              genes  meandiff         sd        z  hgnc
486 ENSG00000167526 10.210689 0.19199110 53.18314 RPL13
194 ENSG00000227507  3.589504 0.06890271 52.09525   LTB
467 ENSG00000140988  8.708400 0.18764717 46.40837  RPS2
110 ENSG00000144713  5.916930 0.12979659 45.58617 RPL32
9   ENSG00000142676  5.859918 0.12964041 45.20133 RPL11
289 ENSG00000137154  7.297226 0.16245776 44.91768  RPS6</code></pre>
<pre class="r"><code>head(sce@int_metadata$hippo$diffexp$result_table[[3]]) #round 3: regulatory T cells</code></pre>
<pre><code>              genes  meandiff         sd         z          hgnc
54  ENSG00000172116 0.3270666 0.02264913 14.440578          CD8B
330 ENSG00000126353 0.2942714 0.02739772 10.740725          CCR7
245 ENSG00000256039 0.1476583 0.01410074 10.471670 RP11-291B21.2
410 ENSG00000160307 0.1486600 0.01449627 10.255055         S100B
338 ENSG00000173762 0.4176998 0.04273547  9.774078           CD7
270 ENSG00000122026 1.5208182 0.16421077  9.261379         RPL21</code></pre>
</div>
<div id="session-info" class="section level2">
<h2>Session Info</h2>
<pre class="r"><code>sessionInfo()</code></pre>
<pre><code>R version 3.6.1 (2019-07-05)
Platform: x86_64-apple-darwin15.6.0 (64-bit)
Running under: macOS Sierra 10.12.6

Matrix products: default
BLAS:   /Library/Frameworks/R.framework/Versions/3.6/Resources/lib/libRblas.0.dylib
LAPACK: /Library/Frameworks/R.framework/Versions/3.6/Resources/lib/libRlapack.dylib

locale:
[1] en_US.UTF-8/en_US.UTF-8/en_US.UTF-8/C/en_US.UTF-8/en_US.UTF-8

attached base packages:
[1] parallel  stats4    stats     graphics  grDevices utils     datasets 
[8] methods   base     

other attached packages:
 [1] Matrix_1.2-17               SingleCellExperiment_1.8.0 
 [3] SummarizedExperiment_1.16.1 DelayedArray_0.12.1        
 [5] BiocParallel_1.20.1         matrixStats_0.55.0         
 [7] Biobase_2.46.0              GenomicRanges_1.38.0       
 [9] GenomeInfoDb_1.22.0         IRanges_2.20.1             
[11] S4Vectors_0.24.1            BiocGenerics_0.32.0        
[13] HIPPO_0.1.0                

loaded via a namespace (and not attached):
 [1] Rcpp_1.0.2             lattice_0.20-38        assertthat_0.2.1      
 [4] rprojroot_1.3-2        digest_0.6.21          RSpectra_0.15-0       
 [7] R6_2.4.0               plyr_1.8.4             backports_1.1.5       
[10] evaluate_0.14          ggplot2_3.2.1          pillar_1.4.2          
[13] zlibbioc_1.32.0        rlang_0.4.1            lazyeval_0.2.2        
[16] data.table_1.12.6      irlba_2.3.3            whisker_0.4           
[19] reticulate_1.13        rmarkdown_2.0          labeling_0.3          
[22] Rtsne_0.15             stringr_1.4.0          RCurl_1.95-4.12       
[25] munsell_0.5.0          umap_0.2.3.1           compiler_3.6.1        
[28] httpuv_1.5.2           xfun_0.11              pkgconfig_2.0.3       
[31] askpass_1.1            htmltools_0.4.0        openssl_1.4.1         
[34] tidyselect_0.2.5       tibble_2.1.3           gridExtra_2.3         
[37] GenomeInfoDbData_1.2.2 workflowr_1.5.0        crayon_1.3.4          
[40] dplyr_0.8.3            later_1.0.0            bitops_1.0-6          
[43] grid_3.6.1             jsonlite_1.6           gtable_0.3.0          
[46] git2r_0.26.1           magrittr_1.5           scales_1.0.0          
[49] stringi_1.4.3          XVector_0.26.0         reshape2_1.4.3        
[52] fs_1.3.1               promises_1.1.0         tools_3.6.1           
[55] glue_1.3.1             purrr_0.3.3            yaml_2.2.0            
[58] colorspace_1.4-1       knitr_1.26            </code></pre>
<br>
<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-sessioninfo" data-toggle="collapse" data-target="#workflowr-sessioninfo" style="display: block;">
<span class="glyphicon glyphicon-wrench" aria-hidden="true"></span> Session information
</button>
</p>
<div id="workflowr-sessioninfo" class="collapse">
<pre class="r"><code>sessionInfo()</code></pre>
<pre><code>R version 3.6.1 (2019-07-05)
Platform: x86_64-apple-darwin15.6.0 (64-bit)
Running under: macOS Sierra 10.12.6

Matrix products: default
BLAS:   /Library/Frameworks/R.framework/Versions/3.6/Resources/lib/libRblas.0.dylib
LAPACK: /Library/Frameworks/R.framework/Versions/3.6/Resources/lib/libRlapack.dylib

locale:
[1] en_US.UTF-8/en_US.UTF-8/en_US.UTF-8/C/en_US.UTF-8/en_US.UTF-8

attached base packages:
[1] parallel  stats4    stats     graphics  grDevices utils     datasets 
[8] methods   base     

other attached packages:
 [1] Matrix_1.2-17               SingleCellExperiment_1.8.0 
 [3] SummarizedExperiment_1.16.1 DelayedArray_0.12.1        
 [5] BiocParallel_1.20.1         matrixStats_0.55.0         
 [7] Biobase_2.46.0              GenomicRanges_1.38.0       
 [9] GenomeInfoDb_1.22.0         IRanges_2.20.1             
[11] S4Vectors_0.24.1            BiocGenerics_0.32.0        
[13] HIPPO_0.1.0                

loaded via a namespace (and not attached):
 [1] Rcpp_1.0.2             lattice_0.20-38        assertthat_0.2.1      
 [4] rprojroot_1.3-2        digest_0.6.21          RSpectra_0.15-0       
 [7] R6_2.4.0               plyr_1.8.4             backports_1.1.5       
[10] evaluate_0.14          ggplot2_3.2.1          pillar_1.4.2          
[13] zlibbioc_1.32.0        rlang_0.4.1            lazyeval_0.2.2        
[16] data.table_1.12.6      irlba_2.3.3            whisker_0.4           
[19] reticulate_1.13        rmarkdown_2.0          labeling_0.3          
[22] Rtsne_0.15             stringr_1.4.0          RCurl_1.95-4.12       
[25] munsell_0.5.0          umap_0.2.3.1           compiler_3.6.1        
[28] httpuv_1.5.2           xfun_0.11              pkgconfig_2.0.3       
[31] askpass_1.1            htmltools_0.4.0        openssl_1.4.1         
[34] tidyselect_0.2.5       tibble_2.1.3           gridExtra_2.3         
[37] GenomeInfoDbData_1.2.2 workflowr_1.5.0        crayon_1.3.4          
[40] dplyr_0.8.3            later_1.0.0            bitops_1.0-6          
[43] grid_3.6.1             jsonlite_1.6           gtable_0.3.0          
[46] git2r_0.26.1           magrittr_1.5           scales_1.0.0          
[49] stringi_1.4.3          XVector_0.26.0         reshape2_1.4.3        
[52] fs_1.3.1               promises_1.1.0         tools_3.6.1           
[55] glue_1.3.1             purrr_0.3.3            yaml_2.2.0            
[58] colorspace_1.4-1       knitr_1.26            </code></pre>
</div>
</div>


<!-- Adjust MathJax settings so that all math formulae are shown using
TeX fonts only; see
http://docs.mathjax.org/en/latest/configuration.html.  This will make
the presentation more consistent at the cost of the webpage sometimes
taking slightly longer to load. Note that this only works because the
footer is added to webpages before the MathJax javascript. -->
<script type="text/x-mathjax-config">
  MathJax.Hub.Config({
    "HTML-CSS": { availableFonts: ["TeX"] }
  });
</script>



</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.header').parent('thead').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- tabsets -->

<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});

$(document).ready(function () {
  $('.tabset-dropdown > .nav-tabs > li').click(function () {
    $(this).parent().toggleClass('nav-tabs-open')
  });
});
</script>

<!-- code folding -->


<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
