<!DOCTYPE html>

<html>

<head>

<meta charset="utf-8" />
<meta name="generator" content="pandoc" />
<meta http-equiv="X-UA-Compatible" content="IE=EDGE" />




<title>Example2: PBMC68K</title>

<script src="site_libs/jquery-1.11.3/jquery.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/cosmo.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<link href="site_libs/highlightjs-9.12.0/textmate.css" rel="stylesheet" />
<script src="site_libs/highlightjs-9.12.0/highlight.js"></script>

<style type="text/css">code{white-space: pre;}</style>
<style type="text/css">
  pre:not([class]) {
    background-color: white;
  }
</style>
<script type="text/javascript">
if (window.hljs) {
  hljs.configure({languages: []});
  hljs.initHighlightingOnLoad();
  if (document.readyState && document.readyState === "complete") {
    window.setTimeout(function() { hljs.initHighlighting(); }, 0);
  }
}
</script>



<style type="text/css">
h1 {
  font-size: 34px;
}
h1.title {
  font-size: 38px;
}
h2 {
  font-size: 30px;
}
h3 {
  font-size: 24px;
}
h4 {
  font-size: 18px;
}
h5 {
  font-size: 16px;
}
h6 {
  font-size: 12px;
}
.table th:not([align]) {
  text-align: left;
}
</style>




<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
code {
  color: inherit;
  background-color: rgba(0, 0, 0, 0.04);
}
img {
  max-width:100%;
}
.tabbed-pane {
  padding-top: 12px;
}
.html-widget {
  margin-bottom: 20px;
}
button.code-folding-btn:focus {
  outline: none;
}
summary {
  display: list-item;
}
</style>


<style type="text/css">
/* padding for bootstrap navbar */
body {
  padding-top: 51px;
  padding-bottom: 40px;
}
/* offset scroll position for anchor links (for fixed navbar)  */
.section h1 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h2 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h3 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h4 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h5 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h6 {
  padding-top: 56px;
  margin-top: -56px;
}
.dropdown-submenu {
  position: relative;
}
.dropdown-submenu>.dropdown-menu {
  top: 0;
  left: 100%;
  margin-top: -6px;
  margin-left: -1px;
  border-radius: 0 6px 6px 6px;
}
.dropdown-submenu:hover>.dropdown-menu {
  display: block;
}
.dropdown-submenu>a:after {
  display: block;
  content: " ";
  float: right;
  width: 0;
  height: 0;
  border-color: transparent;
  border-style: solid;
  border-width: 5px 0 5px 5px;
  border-left-color: #cccccc;
  margin-top: 5px;
  margin-right: -10px;
}
.dropdown-submenu:hover>a:after {
  border-left-color: #ffffff;
}
.dropdown-submenu.pull-left {
  float: none;
}
.dropdown-submenu.pull-left>.dropdown-menu {
  left: -100%;
  margin-left: 10px;
  border-radius: 6px 0 6px 6px;
}
</style>

<script>
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark it active
  menuAnchor.parent().addClass('active');

  // if it's got a parent navbar menu mark it active as well
  menuAnchor.closest('li.dropdown').addClass('active');
});
</script>

<!-- tabsets -->

<style type="text/css">
.tabset-dropdown > .nav-tabs {
  display: inline-table;
  max-height: 500px;
  min-height: 44px;
  overflow-y: auto;
  background: white;
  border: 1px solid #ddd;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs > li.active:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li.active:before {
  content: "&#xe258;";
  border: none;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs > li.active {
  display: block;
}

.tabset-dropdown > .nav-tabs > li > a,
.tabset-dropdown > .nav-tabs > li > a:focus,
.tabset-dropdown > .nav-tabs > li > a:hover {
  border: none;
  display: inline-block;
  border-radius: 4px;
  background-color: transparent;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li {
  display: block;
  float: none;
}

.tabset-dropdown > .nav-tabs > li {
  display: none;
}
</style>

<!-- code folding -->




</head>

<body>


<div class="container-fluid main-container">




<div class="navbar navbar-default  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">HIPPO</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="index.html">Home</a>
</li>
<li>
  <a href="about.html">About</a>
</li>
<li>
  <a href="example.html">Example1</a>
</li>
<li>
  <a href="example2.html">Example2</a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<!-- Add a small amount of space between sections. -->
<style type="text/css">
div.section {
  padding-top: 12px;
}
</style>

<div class="fluid-row" id="header">



<h1 class="title toc-ignore">Example2: PBMC68K</h1>

</div>


<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-report" data-toggle="collapse" data-target="#workflowr-report">
<span class="glyphicon glyphicon-list" aria-hidden="true"></span> workflowr <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span>
</button>
</p>
<div id="workflowr-report" class="collapse">
<ul class="nav nav-tabs">
<li class="active">
<a data-toggle="tab" href="#summary">Summary</a>
</li>
<li>
<a data-toggle="tab" href="#checks"> Checks <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> </a>
</li>
<li>
<a data-toggle="tab" href="#versions">Past versions</a>
</li>
</ul>
<div class="tab-content">
<div id="summary" class="tab-pane fade in active">
<p>
<strong>Last updated:</strong> 2020-01-31
</p>
<p>
<strong>Checks:</strong> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> 7 <span class="glyphicon glyphicon-exclamation-sign text-danger" aria-hidden="true"></span> 0
</p>
<p>
<strong>Knit directory:</strong> <code>HIPPO/</code> <span class="glyphicon glyphicon-question-sign" aria-hidden="true" title="This is the local directory in which the code in this file was executed."> </span>
</p>
<p>
This reproducible <a href="http://rmarkdown.rstudio.com">R Markdown</a> analysis was created with <a
  href="https://github.com/jdblischak/workflowr">workflowr</a> (version 1.5.0). The <em>Checks</em> tab describes the reproducibility checks that were applied when the results were created. The <em>Past versions</em> tab lists the development history.
</p>
<hr>
</div>
<div id="checks" class="tab-pane fade">
<div id="workflowr-checks" class="panel-group">
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRMarkdownfilestronguptodate"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>R Markdown file:</strong> up-to-date </a>
</p>
</div>
<div id="strongRMarkdownfilestronguptodate" class="panel-collapse collapse">
<div class="panel-body">
<p>Great! Since the R Markdown file has been committed to the Git repository, you know the exact version of the code that produced these results.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongEnvironmentstrongempty"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Environment:</strong> empty </a>
</p>
</div>
<div id="strongEnvironmentstrongempty" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! The global environment was empty. Objects defined in the global environment can affect the analysis in your R Markdown file in unknown ways. For reproduciblity it’s best to always run the code in an empty environment.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSeedstrongcodesetseed20200117code"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Seed:</strong> <code>set.seed(20200117)</code> </a>
</p>
</div>
<div id="strongSeedstrongcodesetseed20200117code" class="panel-collapse collapse">
<div class="panel-body">
<p>The command <code>set.seed(20200117)</code> was run prior to running the code in the R Markdown file. Setting a seed ensures that any results that rely on randomness, e.g. subsampling or permutations, are reproducible.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSessioninformationstrongrecorded"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Session information:</strong> recorded </a>
</p>
</div>
<div id="strongSessioninformationstrongrecorded" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Recording the operating system, R version, and package versions is critical for reproducibility.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongCachestrongnone"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Cache:</strong> none </a>
</p>
</div>
<div id="strongCachestrongnone" class="panel-collapse collapse">
<div class="panel-body">
<p>Nice! There were no cached chunks for this analysis, so you can be confident that you successfully produced the results during this run.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongFilepathsstrongrelative"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>File paths:</strong> relative </a>
</p>
</div>
<div id="strongFilepathsstrongrelative" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Using relative paths to the files within your workflowr project makes it easier to run your code on other machines.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRepositoryversionstrongahrefhttpsgithubcomtk382HIPPOtree6fb0e1274725aafd78d3ce17c0db3f08507c3704targetblank6fb0e12a"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Repository version:</strong> <a href="https://github.com/tk382/HIPPO/tree/6fb0e1274725aafd78d3ce17c0db3f08507c3704" target="_blank">6fb0e12</a> </a>
</p>
</div>
<div id="strongRepositoryversionstrongahrefhttpsgithubcomtk382HIPPOtree6fb0e1274725aafd78d3ce17c0db3f08507c3704targetblank6fb0e12a" class="panel-collapse collapse">
<div class="panel-body">
<p>
Great! You are using Git for version control. Tracking code development and connecting the code version to the results is critical for reproducibility. The version displayed above was the version of the Git repository at the time these results were generated. <br><br> Note that you need to be careful to ensure that all relevant files for the analysis have been committed to Git prior to generating the results (you can use <code>wflow_publish</code> or <code>wflow_git_commit</code>). workflowr only checks the R Markdown file, but you know if there are other scripts or data files that it depends on. Below is the status of the Git repository when the results were generated:
</p>
<pre><code>
Ignored files:
    Ignored:    .Rhistory
    Ignored:    .Rproj.user/
    Ignored:    R/.Rhistory
    Ignored:    analysis/figure/

Untracked files:
    Untracked:  HIPPO_0.1.0.tar.gz
    Untracked:  make_sce_zhengmix4eq.R
    Untracked:  sce_full_Zhengmix4eq.rds

</code></pre>
<p>
Note that any generated files, e.g. HTML, png, CSS, etc., are not included in this status report because it is ok for generated content to have uncommitted changes.
</p>
</div>
</div>
</div>
</div>
<hr>
</div>
<div id="versions" class="tab-pane fade">

<p>
These are the previous versions of the R Markdown and HTML files. If you’ve configured a remote Git repository (see <code>?wflow_git_remote</code>), click on the hyperlinks in the table below to view them.
</p>
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
File
</th>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
<th>
Message
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/tk382/HIPPO/blob/7f76bf4e1ebe5920211debffc2af916ba09762b1/analysis/example2.Rmd" target="_blank">7f76bf4</a>
</td>
<td>
tk382
</td>
<td>
2020-01-31
</td>
<td>
Matrix error in hippo and remove irrelevant comment in example2
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/tk382/HIPPO/blob/e6ce1d842403701edaca3e53e480c071ddf917d6/analysis/example2.Rmd" target="_blank">e6ce1d8</a>
</td>
<td>
tk382
</td>
<td>
2020-01-30
</td>
<td>
workflowr::wflow_publish(“analysis/example2.Rmd”)
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/tk382/HIPPO/323564eacc07aeb843cb260da31fc240330fd63c/docs/example2.html" target="_blank">323564e</a>
</td>
<td>
tk382
</td>
<td>
2020-01-22
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/tk382/HIPPO/blob/3531e4b053947044296239ba485475125898c919/analysis/example2.Rmd" target="_blank">3531e4b</a>
</td>
<td>
tk382
</td>
<td>
2020-01-22
</td>
<td>
wflow_publish(“analysis/example2.Rmd”,
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/tk382/HIPPO/8809a3d163a5bb6985a6655b5e6008d6e13054e7/docs/example2.html" target="_blank">8809a3d</a>
</td>
<td>
tk382
</td>
<td>
2020-01-22
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/tk382/HIPPO/blob/e5f797b66f62043a381fa83a7489ae9f5953279b/analysis/example2.Rmd" target="_blank">e5f797b</a>
</td>
<td>
tk382
</td>
<td>
2020-01-22
</td>
<td>
wflow_publish(“analysis/example2.Rmd”, republish = TRUE)
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/tk382/HIPPO/75f668f46d4bed083afa1a218e723965db5a8c31/docs/example2.html" target="_blank">75f668f</a>
</td>
<td>
tk382
</td>
<td>
2020-01-17
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/tk382/HIPPO/blob/b78600d87357063b7b5b56f3a70876c2ab1fa2af/analysis/example2.Rmd" target="_blank">b78600d</a>
</td>
<td>
tk382
</td>
<td>
2020-01-17
</td>
<td>
wflow_publish(c(“analysis/example.Rmd”, “analysis/example2.Rmd”,
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/tk382/HIPPO/blob/b404eaefa44b5c24a761b7d5e09c10bf821a8eda/analysis/example2.Rmd" target="_blank">b404eae</a>
</td>
<td>
tk382
</td>
<td>
2020-01-17
</td>
<td>
add workflow analysis files
</td>
</tr>
</tbody>
</table>
</div>
<hr>
</div>
</div>
</div>
<div id="read-the-data" class="section level2">
<h2>Read the data</h2>
<p>This data set is available in 10X <a href="https://support.10xgenomics.com/single-cell-gene-expression/datasets/1.1.0/fresh_68k_pbmc_donor_a">website</a>. The format is Matrix, so we need Matrix package for this particular example.</p>
<pre class="r"><code>library(Matrix)
X = readMM(&quot;/Volumes/tae/Work/SC/data/10X_68K/filtered_matrix/hg19/matrix.mtx&quot;)
rowdata =  read.table(&quot;/Volumes/tae/Work/SC/data/10X_68K/filtered_matrix/hg19/genes.tsv&quot;) # this shows both ENSG IDs and HGNC symbols
coldata = data.table::setDF(data.table::fread(&quot;/Volumes/tae/Work/SC/data/10X_68K/filtered_matrix/hg19/barocdes_annotation.tsv&quot;)) # this shows barcode information
rownames(X) = rowdata$V1
colnames(X) = coldata$barcodes</code></pre>
<p>HIPPO also works with a sparse matrix object (dgCMatrix from Matrix package) as well, but the visualization package UMAP and Rtsne does not work with it well as of now.</p>
<p>To expedite the process, I sampled 10,000 cells from 68,000 cells.</p>
<pre class="r"><code>set.seed(20200117)
cells = sample(1:ncol(X), size = 10000)
X = X[, cells]
coldata = coldata[cells, ]
rm(cells)</code></pre>
<p>Then create the SingleCellExperiment object.</p>
<pre class="r"><code>sce = SingleCellExperiment(assays = list(counts = X),
                           rowData = rowdata,
                           colData = coldata)</code></pre>
</div>
<div id="diagnostic-plot" class="section level2">
<h2>Diagnostic Plot</h2>
<p>This shows how noisy our initial data is.</p>
<pre class="r"><code>hippo_diagnostic_plot(sce)</code></pre>
<p><img src="figure/example2.Rmd/unnamed-chunk-5-1.png" width="672" style="display: block; margin: auto;" /></p>
</div>
<div id="feature-selection-and-hierarchical-clustering" class="section level2">
<h2>Feature Selection and Hierarchical Clustering</h2>
<p>This step performs the feature selection and hierarchical clustering.</p>
<pre class="r"><code>sce = hippo(sce, K = 7, z_threshold = 1.5, outlier_proportion = 0.005)</code></pre>
<pre><code>[1] &quot;K = 2..&quot;
[1] &quot;K = 3..&quot;
[1] &quot;K = 4..&quot;
[1] &quot;K = 5..&quot;
[1] &quot;K = 6..&quot;
[1] &quot;K = 7..&quot;</code></pre>
</div>
<div id="visualize-the-selected-features-at-each-round" class="section level2">
<h2>Visualize the selected features at each round</h2>
<pre class="r"><code>ref = data.frame(hgnc = rowData(sce)$V2,
                 ensg = rowData(sce)$V1)
zero_proportion_plot(sce, switch_to_hgnc = TRUE, ref = ref)</code></pre>
<p><img src="figure/example2.Rmd/unnamed-chunk-7-1.png" width="576" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-7-1">
Past versions of unnamed-chunk-7-1.png
</button>
</p>
<div id="fig-unnamed-chunk-7-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/tk382/HIPPO/blob/6fb0e1274725aafd78d3ce17c0db3f08507c3704/docs/figure/example2.Rmd/unnamed-chunk-7-1.png" target="_blank">6fb0e12</a>
</td>
<td>
tk382
</td>
<td>
2020-01-31
</td>
</tr>
<tr>
<td>
<a href="https://github.com/tk382/HIPPO/blob/323564eacc07aeb843cb260da31fc240330fd63c/docs/figure/example2.Rmd/unnamed-chunk-7-1.png" target="_blank">323564e</a>
</td>
<td>
tk382
</td>
<td>
2020-01-22
</td>
</tr>
<tr>
<td>
<a href="https://github.com/tk382/HIPPO/blob/8809a3d163a5bb6985a6655b5e6008d6e13054e7/docs/figure/example2.Rmd/unnamed-chunk-7-1.png" target="_blank">8809a3d</a>
</td>
<td>
tk382
</td>
<td>
2020-01-22
</td>
</tr>
<tr>
<td>
<a href="https://github.com/tk382/HIPPO/blob/75f668f46d4bed083afa1a218e723965db5a8c31/docs/figure/example2.Rmd/unnamed-chunk-7-1.png" target="_blank">75f668f</a>
</td>
<td>
tk382
</td>
<td>
2020-01-17
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
<div id="differential-expression-example" class="section level2">
<h2>Differential Expression Example</h2>
<pre class="r"><code>sce = diffexp(sce, top.n = 5, switch_to_hgnc = TRUE, ref = ref)</code></pre>
<p><img src="figure/example2.Rmd/unnamed-chunk-8-1.png" width="576" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-8-1">
Past versions of unnamed-chunk-8-1.png
</button>
</p>
<div id="fig-unnamed-chunk-8-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/tk382/HIPPO/blob/6fb0e1274725aafd78d3ce17c0db3f08507c3704/docs/figure/example2.Rmd/unnamed-chunk-8-1.png" target="_blank">6fb0e12</a>
</td>
<td>
tk382
</td>
<td>
2020-01-31
</td>
</tr>
<tr>
<td>
<a href="https://github.com/tk382/HIPPO/blob/323564eacc07aeb843cb260da31fc240330fd63c/docs/figure/example2.Rmd/unnamed-chunk-8-1.png" target="_blank">323564e</a>
</td>
<td>
tk382
</td>
<td>
2020-01-22
</td>
</tr>
<tr>
<td>
<a href="https://github.com/tk382/HIPPO/blob/8809a3d163a5bb6985a6655b5e6008d6e13054e7/docs/figure/example2.Rmd/unnamed-chunk-8-1.png" target="_blank">8809a3d</a>
</td>
<td>
tk382
</td>
<td>
2020-01-22
</td>
</tr>
<tr>
<td>
<a href="https://github.com/tk382/HIPPO/blob/75f668f46d4bed083afa1a218e723965db5a8c31/docs/figure/example2.Rmd/unnamed-chunk-8-1.png" target="_blank">75f668f</a>
</td>
<td>
tk382
</td>
<td>
2020-01-17
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code>head(sce@int_metadata$hippo$diffexp$result_table[[1]])</code></pre>
<pre><code>            genes  meandiff        sd         z     hgnc
1 ENSG00000087086 24.418789 0.2026009 120.52656      FTL
2 ENSG00000167996 17.947127 0.1743106 102.96061     FTH1
3 ENSG00000019582 11.701227 0.1382068  84.66461     CD74
4 ENSG00000101439  8.677931 0.1095546  79.21099     CST3
5 ENSG00000204287  8.388013 0.1120801  74.83947  HLA-DRA
6 ENSG00000196126  7.500041 0.1053995  71.15823 HLA-DRB1</code></pre>
<pre class="r"><code>head(sce@int_metadata$hippo$diffexp$result_table[[2]])</code></pre>
<pre><code>            genes  meandiff         sd         z  hgnc
1 ENSG00000227507  4.248654 0.03162331 134.35197   LTB
2 ENSG00000167526 12.674643 0.10782083 117.55282 RPL13
3 ENSG00000140988 10.803393 0.10731299 100.67181  RPS2
4 ENSG00000144713  6.569609 0.07240089  90.73934 RPL32
5 ENSG00000137154  8.130273 0.09333585  87.10773  RPS6
6 ENSG00000142676  6.284986 0.07328654  85.75907 RPL11</code></pre>
<pre class="r"><code>head(sce@int_metadata$hippo$diffexp$result_table[[3]])</code></pre>
<pre><code>            genes  meandiff         sd        z   hgnc
1 ENSG00000008517 2.1099913 0.02557734 82.49455   IL32
2 ENSG00000198851 1.1756362 0.01800683 65.28834   CD3E
3 ENSG00000167286 1.4416552 0.02213200 65.13894   CD3D
4 ENSG00000179144 1.3500114 0.02253260 59.91369 GIMAP7
5 ENSG00000173762 0.7975143 0.01585160 50.31127    CD7
6 ENSG00000166710 7.1010735 0.16801923 42.26346    B2M</code></pre>
<br>
<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-sessioninfo" data-toggle="collapse" data-target="#workflowr-sessioninfo" style="display: block;">
<span class="glyphicon glyphicon-wrench" aria-hidden="true"></span> Session information
</button>
</p>
<div id="workflowr-sessioninfo" class="collapse">
<pre class="r"><code>sessionInfo()</code></pre>
<pre><code>R version 3.6.1 (2019-07-05)
Platform: x86_64-apple-darwin15.6.0 (64-bit)
Running under: macOS Sierra 10.12.6

Matrix products: default
BLAS:   /Library/Frameworks/R.framework/Versions/3.6/Resources/lib/libRblas.0.dylib
LAPACK: /Library/Frameworks/R.framework/Versions/3.6/Resources/lib/libRlapack.dylib

locale:
[1] en_US.UTF-8/en_US.UTF-8/en_US.UTF-8/C/en_US.UTF-8/en_US.UTF-8

attached base packages:
[1] parallel  stats4    stats     graphics  grDevices utils     datasets 
[8] methods   base     

other attached packages:
 [1] Matrix_1.2-17               magrittr_1.5               
 [3] SingleCellExperiment_1.8.0  SummarizedExperiment_1.16.1
 [5] DelayedArray_0.12.1         BiocParallel_1.20.1        
 [7] matrixStats_0.55.0          Biobase_2.46.0             
 [9] GenomicRanges_1.38.0        GenomeInfoDb_1.22.0        
[11] IRanges_2.20.1              S4Vectors_0.24.1           
[13] BiocGenerics_0.32.0         HIPPO_0.1.0                

loaded via a namespace (and not attached):
 [1] tidyselect_0.2.5       xfun_0.11              purrr_0.3.3           
 [4] reshape2_1.4.3         lattice_0.20-38        colorspace_1.4-1      
 [7] htmltools_0.4.0        yaml_2.2.0             rlang_0.4.3           
[10] later_1.0.0            pillar_1.4.3           glue_1.3.1            
[13] GenomeInfoDbData_1.2.2 lifecycle_0.1.0        plyr_1.8.5            
[16] zlibbioc_1.32.0        stringr_1.4.0          munsell_0.5.0         
[19] gtable_0.3.0           workflowr_1.5.0        evaluate_0.14         
[22] labeling_0.3           knitr_1.26             httpuv_1.5.2          
[25] irlba_2.3.3            Rcpp_1.0.3             promises_1.1.0        
[28] scales_1.1.0           backports_1.1.5        XVector_0.26.0        
[31] farver_2.0.3           fs_1.3.1               gridExtra_2.3         
[34] ggplot2_3.2.1          digest_0.6.23          stringi_1.4.5         
[37] ggrepel_0.8.1          dplyr_0.8.3            grid_3.6.1            
[40] rprojroot_1.3-2        tools_3.6.1            bitops_1.0-6          
[43] lazyeval_0.2.2         RCurl_1.95-4.12        tibble_2.1.3          
[46] crayon_1.3.4           whisker_0.4            pkgconfig_2.0.3       
[49] data.table_1.12.6      assertthat_0.2.1       rmarkdown_2.0         
[52] R6_2.4.1               git2r_0.26.1           compiler_3.6.1        </code></pre>
</div>
</div>


<!-- Adjust MathJax settings so that all math formulae are shown using
TeX fonts only; see
http://docs.mathjax.org/en/latest/configuration.html.  This will make
the presentation more consistent at the cost of the webpage sometimes
taking slightly longer to load. Note that this only works because the
footer is added to webpages before the MathJax javascript. -->
<script type="text/x-mathjax-config">
  MathJax.Hub.Config({
    "HTML-CSS": { availableFonts: ["TeX"] }
  });
</script>



</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.header').parent('thead').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- tabsets -->

<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});

$(document).ready(function () {
  $('.tabset-dropdown > .nav-tabs > li').click(function () {
    $(this).parent().toggleClass('nav-tabs-open')
  });
});
</script>

<!-- code folding -->


<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
