<!DOCTYPE html>

<html>

<head>

<meta charset="utf-8" />
<meta name="generator" content="pandoc" />
<meta http-equiv="X-UA-Compatible" content="IE=EDGE" />




<title>Example1: Zhengmix4eq</title>

<script src="site_libs/jquery-1.11.3/jquery.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/cosmo.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<link href="site_libs/highlightjs-9.12.0/textmate.css" rel="stylesheet" />
<script src="site_libs/highlightjs-9.12.0/highlight.js"></script>

<style type="text/css">code{white-space: pre;}</style>
<style type="text/css">
  pre:not([class]) {
    background-color: white;
  }
</style>
<script type="text/javascript">
if (window.hljs) {
  hljs.configure({languages: []});
  hljs.initHighlightingOnLoad();
  if (document.readyState && document.readyState === "complete") {
    window.setTimeout(function() { hljs.initHighlighting(); }, 0);
  }
}
</script>



<style type="text/css">
h1 {
  font-size: 34px;
}
h1.title {
  font-size: 38px;
}
h2 {
  font-size: 30px;
}
h3 {
  font-size: 24px;
}
h4 {
  font-size: 18px;
}
h5 {
  font-size: 16px;
}
h6 {
  font-size: 12px;
}
.table th:not([align]) {
  text-align: left;
}
</style>




<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
code {
  color: inherit;
  background-color: rgba(0, 0, 0, 0.04);
}
img {
  max-width:100%;
}
.tabbed-pane {
  padding-top: 12px;
}
.html-widget {
  margin-bottom: 20px;
}
button.code-folding-btn:focus {
  outline: none;
}
summary {
  display: list-item;
}
</style>


<style type="text/css">
/* padding for bootstrap navbar */
body {
  padding-top: 51px;
  padding-bottom: 40px;
}
/* offset scroll position for anchor links (for fixed navbar)  */
.section h1 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h2 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h3 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h4 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h5 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h6 {
  padding-top: 56px;
  margin-top: -56px;
}
.dropdown-submenu {
  position: relative;
}
.dropdown-submenu>.dropdown-menu {
  top: 0;
  left: 100%;
  margin-top: -6px;
  margin-left: -1px;
  border-radius: 0 6px 6px 6px;
}
.dropdown-submenu:hover>.dropdown-menu {
  display: block;
}
.dropdown-submenu>a:after {
  display: block;
  content: " ";
  float: right;
  width: 0;
  height: 0;
  border-color: transparent;
  border-style: solid;
  border-width: 5px 0 5px 5px;
  border-left-color: #cccccc;
  margin-top: 5px;
  margin-right: -10px;
}
.dropdown-submenu:hover>a:after {
  border-left-color: #ffffff;
}
.dropdown-submenu.pull-left {
  float: none;
}
.dropdown-submenu.pull-left>.dropdown-menu {
  left: -100%;
  margin-left: 10px;
  border-radius: 6px 0 6px 6px;
}
</style>

<script>
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark it active
  menuAnchor.parent().addClass('active');

  // if it's got a parent navbar menu mark it active as well
  menuAnchor.closest('li.dropdown').addClass('active');
});
</script>

<!-- tabsets -->

<style type="text/css">
.tabset-dropdown > .nav-tabs {
  display: inline-table;
  max-height: 500px;
  min-height: 44px;
  overflow-y: auto;
  background: white;
  border: 1px solid #ddd;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs > li.active:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li.active:before {
  content: "&#xe258;";
  border: none;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs > li.active {
  display: block;
}

.tabset-dropdown > .nav-tabs > li > a,
.tabset-dropdown > .nav-tabs > li > a:focus,
.tabset-dropdown > .nav-tabs > li > a:hover {
  border: none;
  display: inline-block;
  border-radius: 4px;
  background-color: transparent;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li {
  display: block;
  float: none;
}

.tabset-dropdown > .nav-tabs > li {
  display: none;
}
</style>

<!-- code folding -->




</head>

<body>


<div class="container-fluid main-container">




<div class="navbar navbar-default  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">HIPPO</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="index.html">Home</a>
</li>
<li>
  <a href="about.html">About</a>
</li>
<li>
  <a href="example.html">Example1</a>
</li>
<li>
  <a href="example2.html">Example2</a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<!-- Add a small amount of space between sections. -->
<style type="text/css">
div.section {
  padding-top: 12px;
}
</style>

<div class="fluid-row" id="header">



<h1 class="title toc-ignore">Example1: Zhengmix4eq</h1>

</div>


<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-report" data-toggle="collapse" data-target="#workflowr-report">
<span class="glyphicon glyphicon-list" aria-hidden="true"></span> workflowr <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span>
</button>
</p>
<div id="workflowr-report" class="collapse">
<ul class="nav nav-tabs">
<li class="active">
<a data-toggle="tab" href="#summary">Summary</a>
</li>
<li>
<a data-toggle="tab" href="#checks"> Checks <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> </a>
</li>
<li>
<a data-toggle="tab" href="#versions">Past versions</a>
</li>
</ul>
<div class="tab-content">
<div id="summary" class="tab-pane fade in active">
<p>
<strong>Last updated:</strong> 2020-01-23
</p>
<p>
<strong>Checks:</strong> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> 7 <span class="glyphicon glyphicon-exclamation-sign text-danger" aria-hidden="true"></span> 0
</p>
<p>
<strong>Knit directory:</strong> <code>HIPPO/</code> <span class="glyphicon glyphicon-question-sign" aria-hidden="true" title="This is the local directory in which the code in this file was executed."> </span>
</p>
<p>
This reproducible <a href="http://rmarkdown.rstudio.com">R Markdown</a> analysis was created with <a
  href="https://github.com/jdblischak/workflowr">workflowr</a> (version 1.5.0). The <em>Checks</em> tab describes the reproducibility checks that were applied when the results were created. The <em>Past versions</em> tab lists the development history.
</p>
<hr>
</div>
<div id="checks" class="tab-pane fade">
<div id="workflowr-checks" class="panel-group">
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRMarkdownfilestronguptodate"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>R Markdown file:</strong> up-to-date </a>
</p>
</div>
<div id="strongRMarkdownfilestronguptodate" class="panel-collapse collapse">
<div class="panel-body">
<p>Great! Since the R Markdown file has been committed to the Git repository, you know the exact version of the code that produced these results.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongEnvironmentstrongempty"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Environment:</strong> empty </a>
</p>
</div>
<div id="strongEnvironmentstrongempty" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! The global environment was empty. Objects defined in the global environment can affect the analysis in your R Markdown file in unknown ways. For reproduciblity it’s best to always run the code in an empty environment.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSeedstrongcodesetseed20200117code"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Seed:</strong> <code>set.seed(20200117)</code> </a>
</p>
</div>
<div id="strongSeedstrongcodesetseed20200117code" class="panel-collapse collapse">
<div class="panel-body">
<p>The command <code>set.seed(20200117)</code> was run prior to running the code in the R Markdown file. Setting a seed ensures that any results that rely on randomness, e.g. subsampling or permutations, are reproducible.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSessioninformationstrongrecorded"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Session information:</strong> recorded </a>
</p>
</div>
<div id="strongSessioninformationstrongrecorded" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Recording the operating system, R version, and package versions is critical for reproducibility.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongCachestrongnone"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Cache:</strong> none </a>
</p>
</div>
<div id="strongCachestrongnone" class="panel-collapse collapse">
<div class="panel-body">
<p>Nice! There were no cached chunks for this analysis, so you can be confident that you successfully produced the results during this run.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongFilepathsstrongrelative"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>File paths:</strong> relative </a>
</p>
</div>
<div id="strongFilepathsstrongrelative" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Using relative paths to the files within your workflowr project makes it easier to run your code on other machines.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRepositoryversionstrongahrefhttpsgithubcomtk382HIPPOtree47c6a6c91cfdf48d19364867279b1a596767334btargetblank47c6a6ca"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Repository version:</strong> <a href="https://github.com/tk382/HIPPO/tree/47c6a6c91cfdf48d19364867279b1a596767334b" target="_blank">47c6a6c</a> </a>
</p>
</div>
<div id="strongRepositoryversionstrongahrefhttpsgithubcomtk382HIPPOtree47c6a6c91cfdf48d19364867279b1a596767334btargetblank47c6a6ca" class="panel-collapse collapse">
<div class="panel-body">
<p>
Great! You are using Git for version control. Tracking code development and connecting the code version to the results is critical for reproducibility. The version displayed above was the version of the Git repository at the time these results were generated. <br><br> Note that you need to be careful to ensure that all relevant files for the analysis have been committed to Git prior to generating the results (you can use <code>wflow_publish</code> or <code>wflow_git_commit</code>). workflowr only checks the R Markdown file, but you know if there are other scripts or data files that it depends on. Below is the status of the Git repository when the results were generated:
</p>
<pre><code>
Ignored files:
    Ignored:    .Rhistory
    Ignored:    .Rproj.user/
    Ignored:    R/.Rhistory

Untracked files:
    Untracked:  make_sce_zhengmix4eq.R
    Untracked:  sce_full_Zhengmix4eq.rds

Unstaged changes:
    Modified:   analysis/example2.Rmd
    Deleted:    analysis/figure/example.Rmd/unnamed-chunk-3-1.png
    Deleted:    analysis/figure/example.Rmd/unnamed-chunk-5-1.png
    Deleted:    analysis/figure/example.Rmd/unnamed-chunk-6-1.png

</code></pre>
<p>
Note that any generated files, e.g. HTML, png, CSS, etc., are not included in this status report because it is ok for generated content to have uncommitted changes.
</p>
</div>
</div>
</div>
</div>
<hr>
</div>
<div id="versions" class="tab-pane fade">

<p>
These are the previous versions of the R Markdown and HTML files. If you’ve configured a remote Git repository (see <code>?wflow_git_remote</code>), click on the hyperlinks in the table below to view them.
</p>
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
File
</th>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
<th>
Message
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/tk382/HIPPO/blob/47c6a6c91cfdf48d19364867279b1a596767334b/analysis/example.Rmd" target="_blank">47c6a6c</a>
</td>
<td>
tk382
</td>
<td>
2020-01-23
</td>
<td>
workflowr::wflow_publish(“analysis/example.Rmd”)
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/tk382/HIPPO/323564eacc07aeb843cb260da31fc240330fd63c/docs/example.html" target="_blank">323564e</a>
</td>
<td>
tk382
</td>
<td>
2020-01-22
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/tk382/HIPPO/8809a3d163a5bb6985a6655b5e6008d6e13054e7/docs/example.html" target="_blank">8809a3d</a>
</td>
<td>
tk382
</td>
<td>
2020-01-22
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/tk382/HIPPO/75f668f46d4bed083afa1a218e723965db5a8c31/docs/example.html" target="_blank">75f668f</a>
</td>
<td>
tk382
</td>
<td>
2020-01-17
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/tk382/HIPPO/blob/b78600d87357063b7b5b56f3a70876c2ab1fa2af/analysis/example.Rmd" target="_blank">b78600d</a>
</td>
<td>
tk382
</td>
<td>
2020-01-17
</td>
<td>
wflow_publish(c(“analysis/example.Rmd”, “analysis/example2.Rmd”,
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/tk382/HIPPO/5d79fe159ab5f5b7ab485602fbd3b41fda38fde4/docs/example.html" target="_blank">5d79fe1</a>
</td>
<td>
tk382
</td>
<td>
2020-01-17
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/tk382/HIPPO/blob/043e60cd4c32cf59c37fed6724e57a2deeba7cd0/analysis/example.Rmd" target="_blank">043e60c</a>
</td>
<td>
tk382
</td>
<td>
2020-01-17
</td>
<td>
wflow_publish(c(“analysis/example.Rmd”))
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/tk382/HIPPO/cb4b8f08924067be279b4e072d732cacf9bdcaf7/docs/example.html" target="_blank">cb4b8f0</a>
</td>
<td>
tk382
</td>
<td>
2020-01-17
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/tk382/HIPPO/blob/660f41fee8874ad271acee45fa555a096065a0f6/analysis/example.Rmd" target="_blank">660f41f</a>
</td>
<td>
tk382
</td>
<td>
2020-01-17
</td>
<td>
wflow_publish(c(“analysis/example.Rmd”))
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/tk382/HIPPO/ed7a3997d76e09840fa9e2abb2cb1d7d9a2291a8/docs/example.html" target="_blank">ed7a399</a>
</td>
<td>
tk382
</td>
<td>
2020-01-17
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/tk382/HIPPO/blob/b404eaefa44b5c24a761b7d5e09c10bf821a8eda/analysis/example.Rmd" target="_blank">b404eae</a>
</td>
<td>
tk382
</td>
<td>
2020-01-17
</td>
<td>
add workflow analysis files
</td>
</tr>
</tbody>
</table>
</div>
<hr>
</div>
</div>
</div>
<div id="read-the-data" class="section level2">
<h2>Read the data</h2>
<p>The data set is available in the following <a href="http://imlspenticton.uzh.ch/robinson_lab/DuoClustering2018/DuoClustering2018.tar.gz">link</a>, where the detailed explanation is available <a href="https://github.com/markrobinsonuzh/scRNAseq_clustering_comparison">here</a>. Note that the file is very large (3.3GB). We use Zhengmix4eq data set. From the SingleCellExperiment object, we extracted the count matrix only (gene x cell) as .rds file, and we also created the reference matrix where the gene names are matched between HGNC symbol and ENSG IDs.</p>
<pre class="r"><code>sce = readRDS(&quot;../sce_Zhengmix4eq.rds&quot;)</code></pre>
<p>Alternatively, you can start from a matrix object.</p>
<pre class="r"><code># X = readRDS(&quot;../zhengmix4eq_counts.rds&quot;)
# sce = SingleCellExperiment(assays = list(counts = X))</code></pre>
</div>
<div id="diagnostic-plot" class="section level2">
<h2>Diagnostic Plot</h2>
<p>This plot shows the zero inflation compared to the expected Poisson line. If most genes don’t align with the black line, it shows that there is cell heterogeneity driving the zero inflation.</p>
<pre class="r"><code>hippo_diagnostic_plot(sce, show_outliers = TRUE, zvalue_thresh = 15)</code></pre>
<p><img src="figure/example.Rmd/unnamed-chunk-3-1.png" width="576" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-3-1">
Past versions of unnamed-chunk-3-1.png
</button>
</p>
<div id="fig-unnamed-chunk-3-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/tk382/HIPPO/blob/75f668f46d4bed083afa1a218e723965db5a8c31/docs/figure/example.Rmd/unnamed-chunk-3-1.png" target="_blank">75f668f</a>
</td>
<td>
tk382
</td>
<td>
2020-01-17
</td>
</tr>
<tr>
<td>
<a href="https://github.com/tk382/HIPPO/blob/ed7a3997d76e09840fa9e2abb2cb1d7d9a2291a8/docs/figure/example.Rmd/unnamed-chunk-3-1.png" target="_blank">ed7a399</a>
</td>
<td>
tk382
</td>
<td>
2020-01-17
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
<div id="feature-selection-and-hierarchical-clustering" class="section level2">
<h2>Feature Selection and Hierarchical Clustering</h2>
<p>HIPPO assumes that the count matrix is placed in <a href="mailto:sce@assays@data$counts">sce@assays@data$counts</a>. Some objects that we found online have the count matrix in <a href="mailto:sce@assays$data$counts">sce@assays$data$counts</a>. In this case, HIPPO will throw an error because it cannot found a count matrix. In this case, you have to create another SingleCellExperiment object to assign the count matrix in the correct slot.</p>
<p>Next, you can run hippo function to do the pre-processing that simutlaneously conducts feature selection and hierarchcial clustering. There are three arguments that help you decide the stopping criterion of clustering procedure.</p>
<p>K is the maximum number of clusters that you want. HIPPO will return the clustering results for all k = 2, 3, …, K, so you can overestimate the number of potential clusters. The default is 10, but users are highly recommended to adjust this.</p>
<p>z_threshold is the feature selection criterion. For each round of hierarchical clustering, hippo will find outlier genes where the z-value of significance is greater than the threshold. For example, if you would like to select genes with p-values less than 0.05, z_threshold would be 1.96. The default threshold is 2, but users can use their discretion to change this value.</p>
<p>outlier_proportion is the number of outlier genes to allow. The default is 0.01 (1%) which means the clustering procedure will automatically stop if there are less than 1% of genes remain as important features. With the example data set, the default choice has empirically worked well.</p>
<pre class="r"><code>set.seed(20191031)
sce = hippo(sce, K = 10, z_threshold = 2, outlier_proportion = 0.01)</code></pre>
<pre><code>[1] &quot;K = 2..&quot;
[1] &quot;K = 3..&quot;
[1] &quot;K = 4..&quot;
[1] &quot;K = 5..&quot;
[1] &quot;not enough important features left; terminating the procedure&quot;</code></pre>
</div>
<div id="dimension-reduction-for-each-round-of-hippo" class="section level2">
<h2>Dimension Reduction for Each Round of HIPPO</h2>
<p>We offer two dimension reduction methods: umap and tsne. And we offer two separate visualization functions.</p>
<pre class="r"><code>sce = dimension_reduction(sce, method=&quot;umap&quot;)
hippo_umap_plot(sce)</code></pre>
<p><img src="figure/example.Rmd/unnamed-chunk-5-1.png" width="576" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-5-1">
Past versions of unnamed-chunk-5-1.png
</button>
</p>
<div id="fig-unnamed-chunk-5-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/tk382/HIPPO/blob/75f668f46d4bed083afa1a218e723965db5a8c31/docs/figure/example.Rmd/unnamed-chunk-5-1.png" target="_blank">75f668f</a>
</td>
<td>
tk382
</td>
<td>
2020-01-17
</td>
</tr>
<tr>
<td>
<a href="https://github.com/tk382/HIPPO/blob/cb4b8f08924067be279b4e072d732cacf9bdcaf7/docs/figure/example.Rmd/unnamed-chunk-5-1.png" target="_blank">cb4b8f0</a>
</td>
<td>
tk382
</td>
<td>
2020-01-17
</td>
</tr>
<tr>
<td>
<a href="https://github.com/tk382/HIPPO/blob/ed7a3997d76e09840fa9e2abb2cb1d7d9a2291a8/docs/figure/example.Rmd/unnamed-chunk-5-1.png" target="_blank">ed7a399</a>
</td>
<td>
tk382
</td>
<td>
2020-01-17
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code>sce = dimension_reduction(sce, method=&quot;tsne&quot;)
hippo_tsne_plot(sce)</code></pre>
<p><img src="figure/example.Rmd/unnamed-chunk-6-1.png" width="576" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-6-1">
Past versions of unnamed-chunk-6-1.png
</button>
</p>
<div id="fig-unnamed-chunk-6-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/tk382/HIPPO/blob/75f668f46d4bed083afa1a218e723965db5a8c31/docs/figure/example.Rmd/unnamed-chunk-6-1.png" target="_blank">75f668f</a>
</td>
<td>
tk382
</td>
<td>
2020-01-17
</td>
</tr>
<tr>
<td>
<a href="https://github.com/tk382/HIPPO/blob/cb4b8f08924067be279b4e072d732cacf9bdcaf7/docs/figure/example.Rmd/unnamed-chunk-6-1.png" target="_blank">cb4b8f0</a>
</td>
<td>
tk382
</td>
<td>
2020-01-17
</td>
</tr>
<tr>
<td>
<a href="https://github.com/tk382/HIPPO/blob/ed7a3997d76e09840fa9e2abb2cb1d7d9a2291a8/docs/figure/example.Rmd/unnamed-chunk-6-1.png" target="_blank">ed7a399</a>
</td>
<td>
tk382
</td>
<td>
2020-01-17
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
<div id="visualize-the-selected-features-at-each-round" class="section level2">
<h2>Visualize the selected features at each round</h2>
<p>This function shows how the zero-inflation decreases as HIPPO proceeds in the clustering. This function has arguments called switch_to_hgnc and ref. These aim to provide the users an option to change the gene names from ENSG IDs to HGNC symbols for ease of understanding. Many SingleCellExperiment objects have such data embedded in rowData(sce). Users can create a data frame with ensg and hgnc columns for the genes, and HIPPO will automatically switch the row names of the count matrix from ENSG IDs to HGNC symbols. The default is set to FALSE, assuming that the row names are already HGNC symbols.</p>
<pre class="r"><code>ref = data.frame(hgnc = rowData(sce)$symbol,
                 ensg = rowData(sce)$id)
head(ref)</code></pre>
<pre><code>           hgnc            ensg
1    AL627309.1 ENSG00000237683
2 RP11-206L10.2 ENSG00000228327
3 RP11-206L10.9 ENSG00000237491
4     LINC00115 ENSG00000225880
5        FAM41C ENSG00000230368
6        SAMD11 ENSG00000187634</code></pre>
<pre class="r"><code>zero_proportion_plot(sce, switch_to_hgnc = TRUE, ref = ref)</code></pre>
<pre><code>                 gene gene_mean  gene_var zero_proportion celltype samplesize
237   ENSG00000142676  9.936378 50.375827     0.009748589        2       1949
278   ENSG00000169442  2.243202  8.651295     0.387891226        2       1949
476   ENSG00000142937  6.837866 28.607169     0.031298102        2       1949
658   ENSG00000122406  3.364802 13.178452     0.250897896        2       1949
918   ENSG00000163220  4.158030 36.833330     0.524884556        2       1949
2371  ENSG00000142676  9.936378 50.375827     0.009748589        2       1949
2781  ENSG00000169442  2.243202  8.651295     0.387891226        2       1949
4761  ENSG00000142937  6.837866 28.607169     0.031298102        2       1949
6581  ENSG00000122406  3.364802 13.178452     0.250897896        2       1949
9181  ENSG00000163220  4.158030 36.833330     0.524884556        2       1949
2372  ENSG00000142676  9.936378 50.375827     0.009748589        2       1949
2782  ENSG00000169442  2.243202  8.651295     0.387891226        2       1949
4762  ENSG00000142937  6.837866 28.607169     0.031298102        2       1949
6582  ENSG00000122406  3.364802 13.178452     0.250897896        2       1949
9182  ENSG00000163220  4.158030 36.833330     0.524884556        2       1949
11821 ENSG00000254709  1.759960 83.169641     0.731075697        2       1004
29517 ENSG00000147604  5.527309 19.331013     0.081429990        1       1007
33131 ENSG00000008517  3.913605 16.186365     0.198609732        1       1007
37044 ENSG00000163220  8.566138 38.199276     0.022222222        5        945
37046 ENSG00000143546  7.973545 38.059681     0.043386243        5        945
       expected_pi           se minus_logp  zvalue selected_feature K   hgnc
237   4.839466e-05 0.0001576237        500 31.4843             TRUE 2  RPL11
278   1.061454e-01 0.0069793904        500 31.4843             TRUE 2   CD52
476   1.072665e-03 0.0007417071        500 31.4843             TRUE 2   RPS8
658   3.457771e-02 0.0041399033        500 31.4843             TRUE 2   RPL5
918   1.564235e-02 0.0028116475        500 31.4843             TRUE 2 S100A9
2371  4.839466e-05 0.0001576237        500 31.4843             TRUE 3  RPL11
2781  1.061454e-01 0.0069793904        500 31.4843             TRUE 3   CD52
4761  1.072665e-03 0.0007417071        500 31.4843             TRUE 3   RPS8
6581  3.457771e-02 0.0041399033        500 31.4843             TRUE 3   RPL5
9181  1.564235e-02 0.0028116475        500 31.4843             TRUE 3 S100A9
2372  4.839466e-05 0.0001576237        500 31.4843             TRUE 4  RPL11
2782  1.061454e-01 0.0069793904        500 31.4843             TRUE 4   CD52
4762  1.072665e-03 0.0007417071        500 31.4843             TRUE 4   RPS8
6582  3.457771e-02 0.0041399033        500 31.4843             TRUE 4   RPL5
9182  1.564235e-02 0.0028116475        500 31.4843            FALSE 4 S100A9
11821 1.721374e-01 0.0119212133        500 31.4843            FALSE 5  IGLL5
29517 3.978652e-03 0.0019849853        500 31.4843             TRUE 5   RPL7
33131 1.997831e-02 0.0044121695        500 31.4843             TRUE 5   IL32
37044 1.905476e-04 0.0004492953        500 31.4843            FALSE 5 S100A9
37046 3.446381e-04 0.0006041966        500 31.4843            FALSE 5 S100A8</code></pre>
<p><img src="figure/example.Rmd/unnamed-chunk-7-1.png" width="576" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-7-1">
Past versions of unnamed-chunk-7-1.png
</button>
</p>
<div id="fig-unnamed-chunk-7-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/tk382/HIPPO/blob/75f668f46d4bed083afa1a218e723965db5a8c31/docs/figure/example.Rmd/unnamed-chunk-7-1.png" target="_blank">75f668f</a>
</td>
<td>
tk382
</td>
<td>
2020-01-17
</td>
</tr>
<tr>
<td>
<a href="https://github.com/tk382/HIPPO/blob/cb4b8f08924067be279b4e072d732cacf9bdcaf7/docs/figure/example.Rmd/unnamed-chunk-7-1.png" target="_blank">cb4b8f0</a>
</td>
<td>
tk382
</td>
<td>
2020-01-17
</td>
</tr>
<tr>
<td>
<a href="https://github.com/tk382/HIPPO/blob/ed7a3997d76e09840fa9e2abb2cb1d7d9a2291a8/docs/figure/example.Rmd/unnamed-chunk-7-1.png" target="_blank">ed7a399</a>
</td>
<td>
tk382
</td>
<td>
2020-01-17
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p><img src="figure/example.Rmd/unnamed-chunk-7-2.png" width="576" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-7-2">
Past versions of unnamed-chunk-7-2.png
</button>
</p>
<div id="fig-unnamed-chunk-7-2" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/tk382/HIPPO/blob/75f668f46d4bed083afa1a218e723965db5a8c31/docs/figure/example.Rmd/unnamed-chunk-7-2.png" target="_blank">75f668f</a>
</td>
<td>
tk382
</td>
<td>
2020-01-17
</td>
</tr>
<tr>
<td>
<a href="https://github.com/tk382/HIPPO/blob/cb4b8f08924067be279b4e072d732cacf9bdcaf7/docs/figure/example.Rmd/unnamed-chunk-7-2.png" target="_blank">cb4b8f0</a>
</td>
<td>
tk382
</td>
<td>
2020-01-17
</td>
</tr>
<tr>
<td>
<a href="https://github.com/tk382/HIPPO/blob/ed7a3997d76e09840fa9e2abb2cb1d7d9a2291a8/docs/figure/example.Rmd/unnamed-chunk-7-2.png" target="_blank">ed7a399</a>
</td>
<td>
tk382
</td>
<td>
2020-01-17
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
<div id="differential-expression-example" class="section level2">
<h2>Differential Expression Example</h2>
<p>We also offer a differential expression analysis tool.</p>
<p>This function also has an option to switch the gene names to HGNC symbols. top.n argument lets users choose how many top genes to show in the box plot. The default is 5.</p>
<pre class="r"><code>sce = diffexp(sce, top.n = 5, switch_to_hgnc = TRUE, ref = ref)</code></pre>
<p><img src="figure/example.Rmd/unnamed-chunk-8-1.png" width="576" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-8-1">
Past versions of unnamed-chunk-8-1.png
</button>
</p>
<div id="fig-unnamed-chunk-8-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/tk382/HIPPO/blob/75f668f46d4bed083afa1a218e723965db5a8c31/docs/figure/example.Rmd/unnamed-chunk-8-1.png" target="_blank">75f668f</a>
</td>
<td>
tk382
</td>
<td>
2020-01-17
</td>
</tr>
<tr>
<td>
<a href="https://github.com/tk382/HIPPO/blob/cb4b8f08924067be279b4e072d732cacf9bdcaf7/docs/figure/example.Rmd/unnamed-chunk-8-1.png" target="_blank">cb4b8f0</a>
</td>
<td>
tk382
</td>
<td>
2020-01-17
</td>
</tr>
<tr>
<td>
<a href="https://github.com/tk382/HIPPO/blob/ed7a3997d76e09840fa9e2abb2cb1d7d9a2291a8/docs/figure/example.Rmd/unnamed-chunk-8-1.png" target="_blank">ed7a399</a>
</td>
<td>
tk382
</td>
<td>
2020-01-17
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p>The labels of boxplots are not quite straightfoward, as we look at different cell groups at each round of HIPPO. This must be interepreted simultaneously with the hierarchical clustering plot such as t-SNE plot above.</p>
<p>First, in the first round, K moves from 1 to 2, and the red group is separated. This group is Monocytes in this particular data set. The box plot always shows the “separated” group in the green box, and hence the group 1. The red boxes represent the remaining cells, so groups 2, 3, and 4 combined.</p>
<p>In the second round, K moves from 2 to 3, and as shown in the t-SNE plot, groups 3 and 4 are separated from the group 2 because they’re assigned a new color. Group 2 in this data set is B cells. (Note that this is different from group 2 is separated. Group 2 remains green.) Therefore, in the second round of box plots, the green boxes represent groups 3 and 4, and the red box represents group 2. In this round, the first group of cells have been removed from the samples.</p>
<p>In the last round, K moves from 3 to 4, and group 4 has been assigned a new color of violet. Hence, the green boxes represent group 4, which is Regulatory T cells, while the red boxes represent the remaining cells of group 3: Naive T cells.</p>
<pre class="r"><code>head(sce@int_metadata$hippo$diffexp$result_table[[1]]) #round 1: monocytes</code></pre>
<pre><code>               genes  meandiff         sd         z     hgnc
288  ENSG00000019582 11.146811 0.09138445 121.97711     CD74
73   ENSG00000163220  4.086147 0.04656788  87.74604   S100A9
75   ENSG00000143546  3.825106 0.04478527  85.40990   S100A8
326  ENSG00000204287  5.081744 0.06486717  78.34077  HLA-DRA
328  ENSG00000196126  4.565034 0.05980462  76.33246 HLA-DRB1
1053 ENSG00000087086  7.400703 0.09739490  75.98656      FTL</code></pre>
<pre class="r"><code>head(sce@int_metadata$hippo$diffexp$result_table[[2]]) #round 2: B cells</code></pre>
<pre><code>              genes  meandiff         sd        z hgnc
399 ENSG00000008517 2.6307385 0.03623181 72.60854 IL32
311 ENSG00000167286 1.6506986 0.02870021 57.51522 CD3D
310 ENSG00000198851 1.5618762 0.02791737 55.94640 CD3E
315 ENSG00000139193 0.8038922 0.02002858 40.13726 CD27
70  ENSG00000172116 0.7450100 0.01928112 38.63936 CD8B
149 ENSG00000227507 3.5283579 0.10424241 33.84763  LTB</code></pre>
<pre class="r"><code>head(sce@int_metadata$hippo$diffexp$result_table[[3]]) #round 3: regulatory T cells</code></pre>
<pre><code>              genes meandiff        sd        z  hgnc
75  ENSG00000109475 7.180592 0.1288168 55.74265 RPL34
26  ENSG00000177954 9.368816 0.1688232 55.49483 RPS27
57  ENSG00000144713 9.576198 0.1727974 55.41864 RPL32
116 ENSG00000112306 8.694732 0.1600532 54.32402 RPS12
47  ENSG00000071082 7.166027 0.1321860 54.21170 RPL31
173 ENSG00000177600 8.177438 0.1521632 53.74123 RPLP2</code></pre>
<p>Each round of differential expression test results are also saved in the list of data frames.</p>
<br>
<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-sessioninfo" data-toggle="collapse" data-target="#workflowr-sessioninfo" style="display: block;">
<span class="glyphicon glyphicon-wrench" aria-hidden="true"></span> Session information
</button>
</p>
<div id="workflowr-sessioninfo" class="collapse">
<pre class="r"><code>sessionInfo()</code></pre>
<pre><code>R version 3.6.1 (2019-07-05)
Platform: x86_64-apple-darwin15.6.0 (64-bit)
Running under: macOS Sierra 10.12.6

Matrix products: default
BLAS:   /Library/Frameworks/R.framework/Versions/3.6/Resources/lib/libRblas.0.dylib
LAPACK: /Library/Frameworks/R.framework/Versions/3.6/Resources/lib/libRlapack.dylib

locale:
[1] en_US.UTF-8/en_US.UTF-8/en_US.UTF-8/C/en_US.UTF-8/en_US.UTF-8

attached base packages:
[1] parallel  stats4    stats     graphics  grDevices utils     datasets 
[8] methods   base     

other attached packages:
 [1] SingleCellExperiment_1.8.0  SummarizedExperiment_1.16.1
 [3] DelayedArray_0.12.1         BiocParallel_1.20.1        
 [5] matrixStats_0.55.0          Biobase_2.46.0             
 [7] GenomicRanges_1.38.0        GenomeInfoDb_1.22.0        
 [9] IRanges_2.20.1              S4Vectors_0.24.1           
[11] BiocGenerics_0.32.0         HIPPO_0.1.0                

loaded via a namespace (and not attached):
 [1] ggrepel_0.8.1          Rcpp_1.0.2             lattice_0.20-38       
 [4] assertthat_0.2.1       rprojroot_1.3-2        digest_0.6.21         
 [7] RSpectra_0.15-0        R6_2.4.0               plyr_1.8.4            
[10] backports_1.1.5        evaluate_0.14          ggplot2_3.2.1         
[13] pillar_1.4.2           zlibbioc_1.32.0        rlang_0.4.1           
[16] lazyeval_0.2.2         irlba_2.3.3            whisker_0.4           
[19] Matrix_1.2-17          reticulate_1.13        rmarkdown_2.0         
[22] labeling_0.3           Rtsne_0.15             stringr_1.4.0         
[25] RCurl_1.95-4.12        munsell_0.5.0          umap_0.2.3.1          
[28] compiler_3.6.1         httpuv_1.5.2           xfun_0.11             
[31] pkgconfig_2.0.3        askpass_1.1            htmltools_0.4.0       
[34] openssl_1.4.1          tidyselect_0.2.5       tibble_2.1.3          
[37] gridExtra_2.3          GenomeInfoDbData_1.2.2 workflowr_1.5.0       
[40] crayon_1.3.4           dplyr_0.8.3            later_1.0.0           
[43] bitops_1.0-6           grid_3.6.1             jsonlite_1.6          
[46] gtable_0.3.0           git2r_0.26.1           magrittr_1.5          
[49] scales_1.0.0           stringi_1.4.3          XVector_0.26.0        
[52] reshape2_1.4.3         fs_1.3.1               promises_1.1.0        
[55] tools_3.6.1            glue_1.3.1             purrr_0.3.3           
[58] yaml_2.2.0             colorspace_1.4-1       knitr_1.26            </code></pre>
</div>
</div>


<!-- Adjust MathJax settings so that all math formulae are shown using
TeX fonts only; see
http://docs.mathjax.org/en/latest/configuration.html.  This will make
the presentation more consistent at the cost of the webpage sometimes
taking slightly longer to load. Note that this only works because the
footer is added to webpages before the MathJax javascript. -->
<script type="text/x-mathjax-config">
  MathJax.Hub.Config({
    "HTML-CSS": { availableFonts: ["TeX"] }
  });
</script>



</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.header').parent('thead').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- tabsets -->

<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});

$(document).ready(function () {
  $('.tabset-dropdown > .nav-tabs > li').click(function () {
    $(this).parent().toggleClass('nav-tabs-open')
  });
});
</script>

<!-- code folding -->


<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
