---
title: "Example2: PBMC68K"
output:
  workflowr::wflow_html:
    toc: false
editor_options:
  chunk_output_type: console
---
```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE,
                      message = FALSE,
                      warning = FALSE,
                      comment = NA)
library(HIPPO)
library(SingleCellExperiment)
library(magrittr)
```

## Read the data

This data set is available in 10X [website](https://support.10xgenomics.com/single-cell-gene-expression/datasets/1.1.0/fresh_68k_pbmc_donor_a). The format is Matrix, so we need Matrix package for this particular example.

```{r}
library(Matrix)
X = readMM("/Volumes/tae/Work/SC/data/10X_68K/filtered_matrix/hg19/matrix.mtx")
rowdata =  read.table("/Volumes/tae/Work/SC/data/10X_68K/filtered_matrix/hg19/genes.tsv") # this shows both ENSG IDs and HGNC symbols
coldata = data.table::setDF(data.table::fread("/Volumes/tae/Work/SC/data/10X_68K/filtered_matrix/hg19/barocdes_annotation.tsv")) # this shows barcode information
rownames(X) = rowdata$V1
colnames(X) = coldata$barcodes
```

HIPPO also works with a sparse matrix object (dgCMatrix from Matrix package) as well, but the visualization package UMAP and Rtsne does not work with it well as of now. 

To expedite the process, I sampled 10,000 cells from 68,000 cells.

```{r}
set.seed(20200117)
cells = sample(1:ncol(X), size = 10000)
X = X[, cells]
coldata = coldata[cells, ]
rm(cells)
```

Then create the SingleCellExperiment object.

```{r}
sce = SingleCellExperiment(assays = list(counts = X),
                           rowData = rowdata,
                           colData = coldata)
```

```{r, echo = FALSE, include = FALSE}
rm(X, rowdata, coldata); gc()
```

## Diagnostic Plot

This shows how noisy our initial data is.

```{r}
hippo_diagnostic_plot(sce)
```

## Feature Selection and Hierarchical Clustering

This step performs the feature selection and hierarchical clustering.

```{r}
sce = hippo(sce, K = 7, z_threshold = 1.5, outlier_proportion = 0.005)
```

## Visualize the selected features at each round

```{r,fig.width = 6, fig.height = 4}
ref = data.frame(hgnc = rowData(sce)$V2,
                 ensg = rowData(sce)$V1)
zero_proportion_plot(sce, switch_to_hgnc = TRUE, ref = ref)
```

## Differential Expression Example

```{r, fig.width=6, fig.height=4}
sce = diffexp(sce, top.n = 5, switch_to_hgnc = TRUE, ref = ref)
head(sce@int_metadata$hippo$diffexp$result_table[[1]])
head(sce@int_metadata$hippo$diffexp$result_table[[2]])
head(sce@int_metadata$hippo$diffexp$result_table[[3]])
```


